packages/enzyme/src/Utils.js

```diff
@@ -23,7 +23,9 @@ export function isCustomComponentElement(inst, adapter) {
 }
 
 function propsOfNode(node) {
-  return (node && node.props) || {};
+  return entries((node && node.props) || {})
+    .filter(([, value]) => typeof value !== 'undefined')
+    .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Comment:
This is rather ineffecient, creating as many objects as you have keys, I know this is a testing library, but if you have a lot of tests it starts to add up

Reply:
Sure, I could mutate the accumulator since the overall reduce is pure.

Reply:
Done.